proposed/http-client/http-client.md

```diff
@@ -41,12 +41,11 @@ use Psr\Http\Message\ResponseInterface;
 interface ClientInterface
 {
     /**
-     * Sends a PSR-7 request.
-     *
-     * If a request is sent without any prior configuration, an exception MUST NOT be thrown
-     * when a response is recieved, no matter the HTTP status code.
-     *
-     * If a request is sent without any prior configuration, a HTTP client MUST NOT follow redirects.
+     * Sends a PSR-7 request and returns a PSR-7 response. 
+     * 
+     * Every technically correct HTTP response MUST be returned as is, even if it represents a HTTP 
+     * error response or a redirect instruction. The only exception is 1xx responses which MUST be 
+     * assembled in the HTTP client. 
```

Comment:
I'm not a native speaker, but `assembled` seems like the wrong word to me. I'd word is something like that:

```
/**
 * Sends a PSR-7 request and returns a PSR-7 response.
 *
 * Clients MUST NOT differ in behavior depending on the HTTP status code, i.e. they MUST NOT
 * follow redirects or throw exceptions for 4xx or 5xx responses.
 *
 * Note: 1xx responses are intermediate responses and MUST be handled transparently by the
 * client as defined by the HTTP protocol.
 */
```

Reply:
Thank you. 

I do not really understand what you mean with "MUST be handled **transparently** by the client". 
When I read this it seams like you want to show the 100 response. But I know you are arguing for the opposite.

Reply:
If a client handles things *transparently*, the consumer doesn't know about these things. The term is used for [transparent proxies](https://en.wikipedia.org/wiki/Proxy_server#Transparent_proxy) for example, where the client might not even be able to see that a proxy is used.

Reply:
i agree that you moved on tobias, but please do a pull request to refine the wording here @kelunik. maybe try to formulate the part about transparently in a way that does not assume everybody is aware of the definition (it would make sense to me, but from a natural language point of view its not obvious). the way of putting it in a note is also slightly confusing, as before we indicate to never handle status codes. maybe its easier to understand if we say: 

1xx responses are intermediate responses and MUST be handled transparently by the client as defined by the HTTP protocol. The consumer will not notice whether an intermediate 1xx response happened. For all other HTTP status codes, clients MUST NOT differ in behavior. i.e. they MUST NOT follow redirects nor throw exceptions for 4xx or 5xx responses.